db/dcrpg/internal/stakestmts.go

```diff
@@ -358,25 +358,22 @@ const (
 	DeindexAgendasTableOnBlockTime = `DROP INDEX uix_agendas_block_time;`
 
 	agendaLockinBlock              = `SELECT block_height FROM agendas WHERE locked_in = true AND agenda_id = $4 LIMIT 1`
-	SelectAgendasAgendaVotesByTime = `SELECT block_time AS timestamp,
-			count(CASE WHEN agenda_vote_choice = $1 THEN 1 ELSE NULL END) AS yes,
-			count(CASE WHEN agenda_vote_choice = $2 THEN 1 ELSE NULL END) AS abstain,
-			count(CASE WHEN agenda_vote_choice = $3 THEN 1 ELSE NULL END) AS no,
-			count(*) AS total
-		 FROM agendas
-		WHERE agenda_id = $4
-		  AND block_height <= (` + agendaLockinBlock + `)
+	SelectAgendasAgendaVotesByTime = `SELECT block_time AS timestamp,` + selectAgendasQuery + `
 		GROUP BY timestamp ORDER BY timestamp;`
 
-	SelectAgendasAgendaVotesByHeight = `SELECT block_height,
+	SelectAgendasAgendaVotesByHeight = `SELECT block_height,` + selectAgendasQuery + `
+		GROUP BY block_height ORDER BY block_height;`
+
+	SelectAgendasTotalAgendaVotes = `SELECT ` + selectAgendasQuery + `;`
+
+	selectAgendasQuery = `
 			count(CASE WHEN agenda_vote_choice = $1 THEN 1 ELSE NULL END) AS yes,
 			count(CASE WHEN agenda_vote_choice = $2 THEN 1 ELSE NULL END) AS abstain,
 			count(CASE WHEN agenda_vote_choice = $3 THEN 1 ELSE NULL END) AS no,
 			count(*) AS total
-		 FROM agendas
+		FROM agendas
 		WHERE agenda_id = $4
-		  AND block_height <= (` + agendaLockinBlock + `)
-		GROUP BY block_height;`
```

Comment:
I can see why `GROUP BY block_height ORDER BY block_height` is needed, but it's strange that it ordered properly before.  Just by luck I suppose?

Reply:
Still not sure why it worked before and can't work now.